src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs

```diff
@@ -843,21 +811,17 @@ private static LiftingResult UserDefinedBinaryOperatorCanBeLifted(TypeSymbol lef
 
             // UNDONE: This is a naive quadratic algorithm; there is a linear algorithm that works. Consider using it.
             var candidates = result.Results;
-            for (int i = 0; i < candidates.Count; ++i)
+            for (int i = 1; i < candidates.Count; ++i)
             {
                 if (candidates[i].Kind != OperatorAnalysisResultKind.Applicable)
                 {
                     continue;
                 }
 
                 // Is this applicable operator better than every other applicable method?
-                for (int j = 0; j < candidates.Count; ++j)
+                for (int j = 0; j < i; ++j)
                 {
-                    if (i == j)
-                    {
-                        continue;
-                    }
-                    if (candidates[j].Kind == OperatorAnalysisResultKind.Inapplicable)
+                    if (candidates[j].Kind != OperatorAnalysisResultKind.Applicable)
                     {
                         continue;
                     }
```

Comment:
Actually, I'm not sure this change is correct. It is conceivable that betterness is not transitive, and we have a cycle such that M1 is better than M2, M2 is better than M3, and M3 is better than M1. In such a case this change is not correct (no candidate is better than all others). Similarly, we could have M1 better than M2, M2 better than M3, but not M1 better than M3. In that case depending on the order that the methods appear in the array, we could get different answers (either ambiguous or M1 is best). So I'm going to revert the change on line 824.

However, the change to the loop indices appears correct and can probably be made elsewhere if applicable.

---
In reply to: [78035627](https://github.com/dotnet/roslyn/pull/13648#discussion_r78035627) [](ancestors = 78035627)